Source/Core/UpdaterCommon/UpdaterCommon.cpp

```diff
@@ -0,0 +1,546 @@
+      return false;
+#endif
+  }
+  return true;
+}
+
+bool PerformUpdate(const TodoList& todo, const std::string& install_base_path,
+                   const std::string& content_base_url, const std::string& temp_path)
+{
+  fprintf(log_fp, "Starting download step...\n");
+  if (!DownloadContent(todo.to_download, content_base_url, temp_path))
+    return false;
+  fprintf(log_fp, "Download step completed.\n");
+
+  fprintf(log_fp, "Starting update step...\n");
+  if (!UpdateFiles(todo.to_update, install_base_path, temp_path))
+    return false;
+  fprintf(log_fp, "Update step completed.\n");
+
+  fprintf(log_fp, "Starting deletion step...\n");
+  if (!DeleteObsoleteFiles(todo.to_delete, install_base_path))
+    return false;
+  fprintf(log_fp, "Deletion step completed.\n");
+
+  return true;
+}
+
+void FatalError(const std::string& message)
+{
+  fprintf(log_fp, "%s\n", message.c_str());
+
+  UI::Error(message);
+  UI::Stop();
+}
+
+static std::optional<Manifest> ParseManifest(const std::string& manifest)
+{
+  Manifest parsed;
+  size_t pos = 0;
+
+  while (pos < manifest.size())
+  {
+    size_t filename_end_pos = manifest.find('\t', pos);
+    if (filename_end_pos == std::string::npos)
+    {
+      fprintf(log_fp, "Manifest entry %zu: could not find filename end.\n", parsed.entries.size());
+      return {};
+    }
+    size_t hash_end_pos = manifest.find('\n', filename_end_pos);
+    if (hash_end_pos == std::string::npos)
+    {
+      fprintf(log_fp, "Manifest entry %zu: could not find hash end.\n", parsed.entries.size());
+      return {};
+    }
+
+    std::string filename = manifest.substr(pos, filename_end_pos - pos);
+    std::string hash = manifest.substr(filename_end_pos + 1, hash_end_pos - filename_end_pos - 1);
+    if (hash.size() != 32)
+    {
+      fprintf(log_fp, "Manifest entry %zu: invalid hash: \"%s\".\n", parsed.entries.size(),
+              hash.c_str());
+      return {};
+    }
+
+    Manifest::Hash decoded_hash;
+    if (!HexDecode(hash, decoded_hash.data(), decoded_hash.size()))
+    {
+      fprintf(log_fp, "Manifest entry %zu: invalid hash: \"%s\".\n", parsed.entries.size(),
+              hash.c_str());
+      return {};
+    }
+
+    parsed.entries[filename] = decoded_hash;
+    pos = hash_end_pos + 1;
+  }
+
+  return parsed;
+}
+
+// Not showing a progress bar here because this part is just too quick
+std::optional<Manifest> FetchAndParseManifest(const std::string& url)
+{
+  Common::HttpRequest http;
+
+  Common::HttpRequest::Response resp = http.Get(url);
+  if (!resp)
+  {
+    fprintf(log_fp, "Manifest download failed.\n");
+    return {};
+  }
+
+  std::string contents(reinterpret_cast<char*>(resp->data()), resp->size());
+  std::optional<std::string> maybe_decompressed = GzipInflate(contents);
+  if (!maybe_decompressed)
+    return {};
+  std::string decompressed = std::move(*maybe_decompressed);
+
+  // Split into manifest and signature.
+  size_t boundary = decompressed.rfind("\n\n");
```

Comment:
Turn `"\n\n"` into an `constexpr std::string_view` and make the computations based on that for clarity? Or is that not supported by the compilers you need?

Reply:
We don't have `std::string_view` yet, no.

Reply:
All right. (resolved, I can't do it apparently).